### ISSUE TITLE
Dolt sql-server hangs weirdly when used as the backing store to Altium Designer

### ISSUE THREAD

**Issue by 0x4261756D** [dolthub/dolt#6706]
I have just set up a dolt database to act as a part database for Altium Designer using the MySQL ODBC connector. It only interacts with one view inside the database. Actions like placing parts work most of the time but much slower than before using a Microsoft Access mdb (5s instead of < 1s).  However, the sql-server log says executing the queries themselves only takes ~70ms.
More concerning is the fact that around one in three actions (e.g. placing a part) causes the sql-server (and therefore Altium Designer itself, waiting) to hang indefinitely. When trying to establish new connections (even with other clients) produces no log messages and also hangs those clients. After pressing CTRL+C in the server console multiple log messages appear, unblocking the clients and sometimes even sending back the correct data. Sometimes the server even keeps working normally after that, most of the time it keeps hanging until CTRL-C for new connections.
The queries Altium Designer sends seem pretty basic (mostly `SELECT * FROM view` and config stuff like `SET NAMES`), apart from the `SHOW KEYS FROM view`, but that query is run and instantly returns an error even if the server does not hang.
Unfortunately, I cannot provide the database itself but here is the statement to create the view Altium uses:
SQL statement of the view Altium uses: [statement](https://github.com/dolthub/dolt/files/12703452/query.txt)
Excerpt from the sql-server log: [dolt_local-server_MySql-ODBC_Altium_debug_2023-09-22_excerpt.txt](https://github.com/dolthub/dolt/files/12703285/dolt_local-server_MySql-ODBC_Altium_debug_2023-09-22_excerpt.txt)

**Comment by timsehn** [comment]
f the server is returning queries by the client is not getting them, this seems like something we're doing wrong with ODBC connectors. We may need to pull out Wireshark and see what MySQL sends that is different than what we send.

**Comment by timsehn** [comment]
Link to tool:

https://www.altium.com/altium-designer

**Comment by 0x4261756D** [comment]
dolt version 1.16.2

**Comment by timsehn** [comment]
@fulghum will dig into this Monday with support from @macneale4.

**Comment by 0x4261756D** [comment]
After updating to [v1.16.3](https://github.com/dolthub/dolt/releases/tag/v1.16.3) I was no longer able to reproduce the server hanging.
The slow reaction times still persist though.

**Comment by timsehn** [comment]
Unlikely that we did anything in 1.16.3 that fixed the hanging. It was likely the `dolt sql-server` restart that cleaned up some state.

We need to get it hooked up on our side and see if we can figure out what is up.

**Comment by 0x4261756D** [comment]
When testing with 1.16.2 the `dolt sql-server` was restarted quite often since it kept hanging. Since I cannot repro it anymore I can only guess at the cause and the only change (I noticed) was the update to the new dolt version.

**Comment by fulghum** [comment]
For the issue with `SHOW KEYS FROM view`, I was able to repro a different behavior from Dolt 1.16.0 than what MySQL shows. Dolt 1.16.0 was returning a `table not found` error, where as MySQL just returns an empty result set. 

When I tested with Dolt 1.16.3, Dolt now correctly returns an empty result set, the same as MySQL. `SHOW KEYS FROM view` looks like it was fixed in https://github.com/dolthub/go-mysql-server/commit/ea6de8acfca4dfce7be1f26566742068b6f1e6ee last Friday. 

That seems to explain why you're seeing more stable results with version 1.16.3, since the `SHOW KEYS FROM view` statement will execute the same as MySQL now. It doesn't explain the performance issue of course though. @zachmu is working on getting a repro set up with Altium Designer to see if we can figure out what's going on there.

**Comment by zachmu** [comment]
Hi @0x4261756D,

I managed to the my Altium trial up and running connected to a Dolt database. I don't see the slow reaction times you're talking about -- everything seems pretty snappy, no noticeable lag. Here's how I have the dolt parts library configured:

![image](https://github.com/dolthub/dolt/assets/474773/b4d2019b-6ef3-4458-a48d-5f48777f1743)

I'm running the dolt sql server with `dolt sql-server -l debug`, no special configuration.

Can you tell me a little bit more about your setup? How many parts are in the DB? Does it have the same schema as my parts db (below)?

```sql
parts> show create table parts;
+-------+--------------------------------------------------------------------+
| Table | Create Table                                                       |
+-------+--------------------------------------------------------------------+
| parts | CREATE TABLE `parts` (                                             |
|       |   `ID` int NOT NULL,                                               |
|       |   `Part_Type` varchar(255),                                        |
|       |   `Sub_Type` varchar(255),                                         |
|       |   `Value` varchar(255),                                            |
|       |   `Manufacturer Part Number` varchar(255),                         |
|       |   `Manufacturer` varchar(255),                                     |
|       |   `Manufacturer Part Description` varchar(255),                    |
|       |   `Supplier 1` varchar(255),                                       |
|       |   `Supplier Part Number 1` varchar(255),                           |
|       |   `Package_Case` varchar(255),                                     |
|       |   `RoHS` varchar(255),                                             |
|       |   `Temperature Range` varchar(255),                                |
|       |   `Temperature Coeff` varchar(255),                                |
|       |   `Dielectric` varchar(255),                                       |
|       |   `Tolerance` varchar(255),                                        |
|       |   `Voltage_Rating` varchar(255),                                   |
|       |   `Power_Rating` varchar(255),                                     |
|       |   `Current_Rating` varchar(255),                                   |
|       |   `Comment` varchar(255),                                          |
|       |   `Created By` varchar(255),                                       |
|       |   `Created_Date` varchar(255),                                     |
|       |   `Checked By` varchar(255),                                       |
|       |   `Checked_Date` varchar(255),                                     |
|       |   `Updated By` varchar(255),                                       |
|       |   `Updated_Date` varchar(255),                                     |
|       |   `Part_Status` varchar(255),                                      |
|       |   `Description` varchar(255),                                      |
|       |   `Library Ref` varchar(255),                                      |
|       |   `Library Path` varchar(255),                                     |
|       |   `Footprint Ref` varchar(255),                                    |
|       |   `Footprint Path` varchar(255),                                   |
|       |   `Footprint Ref 2` varchar(255),                                  |
|       |   `Footprint Path 2` varchar(255),                                 |
|       |   `Footprint Ref 3` varchar(255),                                  |
|       |   `Footprint Path 3` varchar(255),                                 |
|       |   `ComponentLink1Description` varchar(255),                        |
|       |   `ComponentLink1URL` varchar(255),                                |
|       |   `ComponentLink2Description` varchar(255),                        |
|       |   `ComponentLink2URL` varchar(255),                                |
|       |   `CAD Model` varchar(255),                                        |
|       |   `Notes` varchar(255),                                            |
|       |   `Sorted_Value` double,                                           |
|       |   `MCN` varchar(255),                                              |
|       |   `Manufacturer 2` varchar(255),                                   |
|       |   `Manufacturer Part Number 2` varchar(255),                       |
|       |   `Manufacturer 3` varchar(255),                                   |
|       |   `Manufacturer Part Number 3` varchar(255),                       |
|       |   `Size / Dimension` varchar(255),                                 |
|       |   `Height - Seated (Max)` varchar(255),                            |
|       |   `Case Code` varchar(255),                                        |
|       |   `Lib Report` varchar(255),                                       |
|       |   PRIMARY KEY (`ID`)                                               |
|       | ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci |
+-------+--------------------------------------------------------------------+
```

If you could give me some hints as to what part of the process is slow, that would be helpful too. I was just dragging parts from the components tab into a schematic. Right-clicking and selecting "Place part" was also very fast.

**Comment by zachmu** [comment]
Offline, we tracked down the slowness to ODBC driver 8.1. 8.0 is what I was testing with, which is why I wasn't experiencing the slowness. I'm going to open another ticket about ODBC support. There are two workaround possible here:

1) Use `127.0.0.1` instead of `localhost` in the connection string, which forces ipv4
2) Use connector version 8.0, listed here as 8.0.32:

https://downloads.mysql.com/archives/c-odbc/

You can have 8.0 and 8.1 installed side by side, just specify the version in the connection string, eg:

`DRIVER={MySQL ODBC 8.0 ANSI Driver};SERVER=localhost;OPTION=3;PORT=3306;DATABASE=parts;USER=root;PWD=;PERSIST SECURITY INFO=True;`

I'm going to close this out since you have a couple different workarounds to resolve your latency issue. Please reopen or file a new issue if you encounter other problems.
